server/service/src/sync/translations/purchase_order.rs

```diff
@@ -0,0 +1,427 @@
+            supplier_discount_amount,
+            donor_link_id,
+            reference,
+            currency_id,
+            foreign_exchange_rate: oms_fields.clone().and_then(|o| o.foreign_exchange_rate),
+            shipping_method: oms_fields.clone().and_then(|o| o.shipping_method),
+            sent_datetime: oms_fields.clone().and_then(|o| o.sent_datetime),
+            contract_signed_datetime: oms_fields.clone().and_then(|o| o.contract_signed_datetime),
+            advance_paid_datetime: oms_fields.clone().and_then(|o| o.advance_paid_datetime),
+            received_at_port_datetime: oms_fields.clone().and_then(|o| o.received_at_port_datetime),
+            expected_delivery_datetime: oms_fields
+                .clone()
+                .and_then(|o| o.expected_delivery_datetime),
+            supplier_agent,
+            authorising_officer_1,
+            authorising_officer_2,
+            additional_instructions,
+            heading_message: oms_fields.and_then(|o| o.heading_message),
+            agent_commission,
+            document_charge,
+            communications_charge,
+            insurance_charge,
+            freight_charge,
+            freight_conditions,
+        };
+        Ok(PullTranslateResult::upsert(result))
+    }
+
+    // TODO add try_translate_from_delete_sync_record
+
+    fn try_translate_to_upsert_sync_record(
+        &self,
+        connection: &StorageConnection,
+        changelog: &ChangelogRow,
+    ) -> Result<PushTranslateResult, anyhow::Error> {
+        let PurchaseOrderRow {
+            id,
+            store_id,
+            user_id,
+            supplier_name_link_id,
+            purchase_order_number,
+            status,
+            created_datetime,
+            confirmed_datetime,
+            delivered_datetime,
+            target_months,
+            comment,
+            supplier_discount_percentage,
+            supplier_discount_amount,
+            donor_link_id,
+            reference,
+            currency_id,
+            foreign_exchange_rate,
+            shipping_method,
+            sent_datetime,
+            contract_signed_datetime,
+            advance_paid_datetime,
+            received_at_port_datetime,
+            expected_delivery_datetime,
+            supplier_agent,
+            authorising_officer_1,
+            authorising_officer_2,
+            additional_instructions,
+            heading_message,
+            agent_commission,
+            document_charge,
+            communications_charge,
+            insurance_charge,
+            freight_charge,
+            freight_conditions,
+        } = PurchaseOrderRepository::new(connection)
+            .query_by_filter(
+                PurchaseOrderFilter::new().id(EqualFilter::equal_to(&changelog.record_id)),
+            )?
+            .pop()
+            .ok_or_else(|| anyhow::anyhow!("Purchase Order not found"))?;
+
+        let oms_fields = if foreign_exchange_rate.is_some()
```

Comment:
I think we can remove this conditional, and just always create the oms_fields data, doesn't matter if it's empty.